plugins/performance-lab/includes/admin/rest-api.php

```diff
@@ -0,0 +1,201 @@
+ * @phpstan-param WP_REST_Request<array<string, mixed>> $request
+ *
+ * @param WP_REST_Request $request Request.
+ * @return WP_REST_Response|WP_Error Response.
+ */
+function perflab_handle_activate_plugin( WP_REST_Request $request ) {
+	require_once ABSPATH . 'wp-admin/includes/plugin.php';
+	require_once ABSPATH . 'wp-admin/includes/plugin-install.php';
+	require_once ABSPATH . 'wp-admin/includes/class-wp-upgrader.php';
+	require_once ABSPATH . 'wp-admin/includes/class-wp-ajax-upgrader-skin.php';
+
+	// Require to make helper functions available.
+	require_once PERFLAB_PLUGIN_DIR_PATH . 'includes/admin/load.php';
+	require_once PERFLAB_PLUGIN_DIR_PATH . 'includes/admin/plugins.php';
+
+	$params = $request->get_json_params();
+
+	// Ensure the 'slug' parameter is present.
+	if ( ! isset( $params['slug'] ) ) {
+		return new WP_Error(
+			'missing_parameter',
+			__( 'Missing required parameter "slug".', 'performance-lab' ),
+			array( 'status' => 400 )
+		);
+	}
+
+	$plugin_slug = perflab_sanitize_plugin_slug( wp_unslash( $params['slug'] ) );
+	if ( null === $plugin_slug ) {
+		return new WP_Error(
+			'invalid_plugin',
+			__( 'Invalid plugin slug provided.', 'performance-lab' ),
+			array( 'status' => 400 )
+		);
+	}
+
+	// Install and activate the plugin and its dependencies.
+	$result = perflab_install_and_activate_plugin( $plugin_slug );
+	if ( is_wp_error( $result ) ) {
+		return new WP_Error(
+			'plugin_activation_failed',
+			$result->get_error_message(),
+			array( 'status' => 500 )
+		);
+	}
+
+	return new WP_REST_Response(
+		array(
+			'success' => true,
+		)
+	);
+}
+
+/**
+ * Handles REST API request to get plugin settings URL.
+ *
+ * @since n.e.x.t
+ * @access private
+ *
+ * @phpstan-param WP_REST_Request<array<string, mixed>> $request
+ *
+ * @param WP_REST_Request $request Request.
+ * @return WP_REST_Response|WP_Error Response.
+ */
+function perflab_handle_get_plugin_settings_url( WP_REST_Request $request ) {
+	require_once ABSPATH . 'wp-admin/includes/plugin.php';
+
+	// Require to make helper functions available.
+	require_once PERFLAB_PLUGIN_DIR_PATH . 'includes/admin/load.php';
+	require_once PERFLAB_PLUGIN_DIR_PATH . 'includes/admin/plugins.php';
+
+	$params = $request->get_json_params();
+
+	// Ensure the 'slug' parameter is present.
+	if ( ! isset( $params['slug'] ) ) {
+		return new WP_Error(
+			'missing_parameter',
+			__( 'Missing required parameter "slug".', 'performance-lab' ),
+			array( 'status' => 400 )
+		);
+	}
+
+	$plugin_slug = perflab_sanitize_plugin_slug( wp_unslash( $params['slug'] ) );
+	if ( null === $plugin_slug ) {
+		return new WP_Error(
+			'invalid_plugin',
+			__( 'Invalid plugin slug provided.', 'performance-lab' ),
+			array( 'status' => 400 )
+		);
+	}
+
+	$plugin_settings_url = perflab_get_plugin_settings_url( $plugin_slug );
+	if ( null === $plugin_settings_url ) {
+		return new WP_REST_Response(
+			array(
+				'success'           => true,
+				'pluginSettingsURL' => false,
+			)
+		);
```

Comment:
I think it would make more sense to return a 404 status code here:

```suggestion
		return new WP_Error( 
			'no_settings_url',
			__( 'No settings URL', 'performance-lab' ),
			array( 'status' => 404 )
		);
```

Reply:
I disagree here. The caller provides the plugin slug, not an identifier for the settings URL. I'd argue a 404 would only be appropriate if the plugin slug was for a plugin that's not active or doesn't exist. But if the plugin is active and just doesn't provide a settings URL, then it shouldn't be a 404, since the endpoint's sole purpose is to retrieve the settings URL for an active plugin.

Reply:
The endpoint is for a settings URL entity though. You provide the slug of the plugin to get the settings URL. This suggestion is intended to be paired with https://github.com/WordPress/performance/pull/1646/files#r1840919242 so that if you `GET /plugin-settings-url/foo` and there is no settings URL for that plugin Foo, then it returns a 404.

Reply:
If we implement the `404` approach browser will log error in console for all the plugins which does not provide settings URL. As the browser logs error for `404` status code returned request even if we wrap it in try catch. How should I approach this issue?

Reply:
Yeah, maybe a 404 is too noisy. The endpoint could just return `null` in that case as opposed to the URL string.

Reply:
```suggestion
		return new WP_REST_Response( null );
```

Reply:
With `null` I have the concern that this may also be the result of a erroring JSON-decode operation.

Maybe we return an object with just `pluginSettingsURL` key, which is either the URL or `false`? That feels reasonable to me.

Reply:
JSON decode where? If in JavaScript then this is `JSON.parse()` throws an error when a parse error occurs. 

I just tried patching the comments endpoint to return bad JSON intentionally:

```diff
--- a/wp-includes/rest-api/endpoints/class-wp-rest-comments-controller.php
+++ b/wp-includes/rest-api/endpoints/class-wp-rest-comments-controller.php
@@ -422,6 +422,8 @@ class WP_REST_Comments_Controller extends WP_REST_Controller {
 	 * @return WP_REST_Response|WP_Error Response object on success, or error object on failure.
 	 */
 	public function get_item( $request ) {
+		die( '{"BADJSON":' );
+
 		$comment = $this->get_comment( $request['id'] );
 		if ( is_wp_error( $comment ) ) {
 			return $comment;
```

Then if I do:

```js
await wp.apiFetch({ path: '/wp/v2/comments/1', method: 'GET' })
```

I get an JS error:

![image](https://github.com/user-attachments/assets/480c230d-11bf-4f91-be76-ff4c1a352eb4)

I don't feel strongly against wrapping the response in an object, however.